[demo] Libtiff incompatibility

hmaarrfk opened the pull request:
Pretty sure you need libtiff to be pinned to 4.0 on windows.

This has been causing me so many headaches....


https://github.com/conda-forge/opencv-feedstock/issues/196

Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

hmaarrfk commented:
On my local computer

```
(m) C:\Users\m>python -c "import cv2"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: DLL load failed: The specified module could not be found.

(m) C:\Users\m>conda list | findstr libtiff
libtiff                   4.1.0                h2e92f26_0    conda-forge

(m) C:\Users\m>conda install libtiff=4.0
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\m\Anaconda3\envs\m

  added / updated specs:
    - libtiff=4.0


The following packages will be DOWNGRADED:

  libtiff                                  4.1.0-h2e92f26_0 --> 4.0.10-h6512ee2_1003
  zstd                                     1.4.3-hd8a0e53_0 --> 1.4.0-hd8a0e53_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

(m) C:\Users\m>python -c "import cv2"

(m) C:\Users\m>conda install libtiff=4.1
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\m\Anaconda3\envs\m

  added / updated specs:
    - libtiff=4.1


The following packages will be UPDATED:

  libtiff                              4.0.10-h6512ee2_1003 --> 4.1.0-h2e92f26_0
  zstd                                     1.4.0-hd8a0e53_0 --> 1.4.3-hd8a0e53_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

(m) C:\Users\m>python -c "import cv2"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: DLL load failed: The specified module could not be found.

```

jakirkham commented on recipe/meta.yaml:
```diff
@@ -203,8 +205,10 @@ outputs:
+        - liblapack * *mkl              # [py==36]
         - liblapack * *mkl              # [py==37]
+        - libtiff 4.0                   # [py==36]
+        - libtiff 4.1                   # [py==37]
```
Do we know why? Would it not be possible to align both to 4.1?

hmaarrfk replied:
I'm just trying to make it fail and to reproduce on a CI.

In the issue, I show the full environment that fails for me.

jakirkham replied:
Thanks for clarifying.

hmaarrfk replied:
I really don't understand what is the difference between my environments. Python 3.7 and libtiff 4.1 and opencv built a while back doesn't work :/

hmaarrfk replied:
I'm really not sure what is different. Maybe opencv is a red herring, and something else is depending on libtiff 4.0

ocefpaf commented:
@hmaarrfk maybe #203 fixed this. Do you mind testing it once the CDN syncs?

hmaarrfk commented:
I'll have to test this in the coming days on my own machines. But for now, I'll close this since a recent PR may have fixes the issue
